veggieseasons/lib/data/model.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:scoped_model/scoped_model.dart';
+import 'package:veggieseasons/data/veggie.dart';
+import 'package:veggieseasons/data/local_veggie_provider.dart';
+
+class AppState extends Model {
+  List<Veggie> veggies;
+
+  AppState() : veggies = LocalVeggieProvider.veggies;
+
+  List<Veggie> get allVeggies => veggies;
+
+  Veggie getVeggie(int id) => veggies.singleWhere((v) => v.id == id);
+
+  List<Veggie> get availableVeggies {
+    Season currentSeason = _getSeasonForDate(DateTime.now());
+    return veggies.where((v) => v.seasons.contains(currentSeason)).toList();
+  }
+
+  List<Veggie> get unavailableVeggies {
+    Season currentSeason = _getSeasonForDate(DateTime.now());
+    return veggies.where((v) => !v.seasons.contains(currentSeason)).toList();
+  }
+
+  List<Veggie> get favoriteVeggies =>
+      veggies.where((v) => v.isFavorite).toList();
+
+  List<Veggie> searchVeggies(String terms) => veggies
+      .where((v) => v.name.toLowerCase().contains(terms.toLowerCase()))
+      .toList();
+
+  void toggleFavorite(int id) {
+    Veggie veggie = getVeggie(id);
+    veggie.isFavorite = !veggie.isFavorite;
+    notifyListeners();
+  }
+
+  Season _getSeasonForDate(DateTime date) {
```

Comment:
This method can be static and [nit] should probably live with Season.

Reply:
Made static. Season is just an enum.